File: C/1_18.c
// 1.18 (**) Extract a slice from a list.

#include <stdio.h>

#define MAX_LENGTH 100

void slice(int *array, int b, int e, int *slice_array)
{
    int i = 0;
    while (b  <= e)
    {
        slice_array[i] = array[b - 1];
        b ++;
        i ++;
    }
}

int main(void)
{
    int arr[MAX_LENGTH] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    int slice_array[MAX_LENGTH];

    int b = 4;
    int e = 5;

    slice(arr, b, e, slice_array);

    for (int i = 0; i < e - b + 1; i++)
    {
        printf("%d\n", slice_array[i]);
    }
    
    return 0;

}

File: Python/1_18.py
# 1.18 (**) Extract a slice from a list.

def _slice(lst, b, e):
    return lst[b - 1 : e]


arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
r_arr = [9, 5, 8, 6, 4, 1, 2 ,5, 9, 8 ,7, 4, 3, 69]

print(_slice(arr, 2, 5))
print(_slice(r_arr, 1, 3))